tracer/test/test-applications/integrations/Samples.InstrumentedTests/Vulnerabilities/EmailHtmlInjection/EmailHtmlInjectionTests.cs

```diff
@@ -0,0 +1,253 @@
+        TestEmailSendCall(() => BuildClient().SendAsync(mailMessage, null));
+        AssertNotVulnerable();
+    }
+
+    [Fact]
+    public void GivenAnEmail_WhenSendAsyncMailMessageNotTaintedVaulesHtml_ThenIsNotVulnerable()
+    {
+        var mailMessage = BuildMailMessage(true, untaintedName, untaintedLastName);
+        TestEmailSendCall(() => BuildClient().SendAsync(mailMessage, null));
+        AssertNotVulnerable();
+    }
+
+    [Fact]
+    public void GivenAnEmail_WhenSendAsyncMailMessageTaintedVaulesNoHtml_ThenIsNotVulnerable()
+    {
+        var mailMessage = BuildMailMessage(false, taintedName, taintedLastName);
+        TestEmailSendCall(() => BuildClient().SendAsync(mailMessage, null));
+        AssertNotVulnerable();
+    }
+
+    // Test public Task SendMailAsync(MailMessage message, CancellationToken cancellationToken)
+
+#if NET5_0_OR_GREATER
+
+    [Fact]
+    public void GivenAnEmail_WhenSendMailAsyncMailMessageCancellationTaintedVaulesHtml_ThenIsVulnerable()
+    {
+        var mailMessage = BuildMailMessage(true, taintedName, taintedLastName);
+        TestEmailSendCall(() => BuildClient().SendMailAsync(mailMessage, new System.Threading.CancellationToken()));
+        AssertVulnerable(emailHtmlInjectionType, "Hi :+-Alice<h1>Hi</h1>-+: :+-Stevens-+:!");
+    }
+
+    [Fact]
+    public void GivenAnEmail_WhenSendMailAsyncMailMessageCancellationTaintedVaulesHtmlEscaped_ThenIsNotVulnerable()
+    {
+        var mailMessage = BuildMailMessage(true, WebUtility.HtmlEncode(taintedName), WebUtility.HtmlEncode(taintedLastName));
+        TestEmailSendCall(() => BuildClient().SendMailAsync(mailMessage, new System.Threading.CancellationToken()));
+        AssertNotVulnerable();
+    }
+
+    [Fact]
+    public void GivenAnEmail_WhenSendMailAsyncMailMessageCancellationTaintedVaulesHtmlEscaped_ThenIsNotVulnerable2()
+    {
+        var mailMessage = BuildMailMessage(true, HttpUtility.HtmlEncode(taintedName), HttpUtility.HtmlEncode(taintedLastName));
+        TestEmailSendCall(() => BuildClient().SendMailAsync(mailMessage, new System.Threading.CancellationToken()));
+        AssertNotVulnerable();
+    }
+
+    [Fact]
+    public void GivenAnEmail_WhenSendMailAsyncMailMessageCancellationNotTaintedVaulesHtml_ThenIsNotVulnerable()
+    {
+        var mailMessage = BuildMailMessage(true, untaintedName, untaintedLastName);
+        TestEmailSendCall(() => BuildClient().SendMailAsync(mailMessage, new System.Threading.CancellationToken()));
+        AssertNotVulnerable();
+    }
+
+    [Fact]
+    public void GivenAnEmail_WhenSendMailAsyncMailMessageCancellationTaintedVaulesNoHtml_ThenIsNotVulnerable()
+    {
+        var mailMessage = BuildMailMessage(false, taintedName, taintedLastName);
+        TestEmailSendCall(() => BuildClient().SendMailAsync(mailMessage, new System.Threading.CancellationToken()));
+        AssertNotVulnerable();
+    }
+
+#endif
+
+    private SmtpClient BuildClient()
```

Comment:
Minor: SmptClient is IDisposable, so maybe calling Dispose would reduce GC overhead and give a small perf boost.

Reply:
Done! Thanks!